server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/PreComputeMessagePreviewListener.java

```diff
@@ -0,0 +1,102 @@
+
+package org.apache.james.jmap.event;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.james.jmap.api.preview.MessagePreviewStore;
+import org.apache.james.jmap.api.preview.Preview;
+import org.apache.james.jmap.draft.model.message.view.MessageFullViewFactory;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.events.Event;
+import org.apache.james.mailbox.events.Group;
+import org.apache.james.mailbox.events.MailboxListener;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageResult;
+import org.parboiled.common.ImmutableList;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+
+import reactor.core.publisher.Mono;
+
+public class PreComputeMessagePreviewListener implements MailboxListener.GroupMailboxListener {
+    public static class PreComputeMessagePreviewListenerGroup extends Group {
+
+    }
+
+    private static final Group GROUP = new PreComputeMessagePreviewListenerGroup();
+
+    private final MessageIdManager messageIdManager;
+    private final MessagePreviewStore messagePreviewStore;
+    private final MessageFullViewFactory messageFullViewFactory;
+
+    private MailboxManager mailboxManager;
+
+    @Inject
+    public PreComputeMessagePreviewListener(MailboxManager mailboxManager, MessageIdManager messageIdManager,
+                                            MessagePreviewStore messagePreviewStore, MessageFullViewFactory messageFullViewFactory) {
+        this.mailboxManager = mailboxManager;
+        this.messageIdManager = messageIdManager;
+        this.messagePreviewStore = messagePreviewStore;
+        this.messageFullViewFactory = messageFullViewFactory;
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public void event(Event event) throws MailboxException {
+        if (event instanceof Added) {
+            MailboxSession session = mailboxManager.createSystemSession(event.getUsername());
+            handleAddedEvent((Added) event, session);
+        }
+    }
+
+    private void handleAddedEvent(Added addedEvent, MailboxSession session) throws MailboxException {
+        List<MessageId> messageIds = addedEvent.getUids()
+            .stream()
+            .map(uid -> addedEvent.getMetaData(uid).getMessageId())
+            .distinct()
+            .collect(Guavate.toImmutableList());
+
+        preComputePreview(messageIds, session);
+    }
+
+    private void preComputePreview(List<MessageId> messagesId, MailboxSession session) throws MailboxException {
+        messageIdManager.getMessages(messagesId, FetchGroupImpl.BODY_CONTENT, session)
```

Comment:
can you confirm that is okay to use `BODY_CONTENT`? I see you are using the FullViewFactory which might requires headers, meatadata to be present

Reply:
IMO we should not use FullViewFactory here.

Reply:
The tests are here to make sure it's ok IMO. The preview is made from the body content, we don't need the rest I think

Reply:
Didn't see the update of comments with @chibenwa 's comment... don't mind my previous comment